x-pack/test/observability_ai_assistant_functional/tests/contextual_insights/index.spec.ts

```diff
@@ -87,9 +90,22 @@ export default function ApiTest({ getService, getPageObjects }: FtrProviderConte
   }
 
   async function navigateToError() {
-    await common.navigateToApp('apm');
-    await browser.get(`${deployment.getHostPort()}/app/apm/services/opbeans-go/errors/`);
-    await testSubjects.click('errorGroupId');
+    await common.navigateToUrl('apm', 'services/opbeans-go/errors/some-expection-key', {
+      shouldUseHashForSubUrl: false,
+    });
+    await header.waitUntilLoadingHasFinished();
+  }
+
+  // open contextual insights component and ensure it was opened
+  async function openContextualInsights() {
+    await retry.try(async () => {
+      await testSubjects.click(ui.pages.contextualInsights.button);
+      const isOpen =
+        (await (
+          await find.byCssSelector(`[aria-controls="${ui.pages.contextualInsights.container}"]`)
+        ).getAttribute('aria-expanded')) === 'true';
+      expect(isOpen).to.be(true);
+    });
```

Comment:
This was the culprit of the failure. The accordion was clicked before it was ready/interactive. This is somewhat ugly but feels robust.

Reply:
nit: i always found `retry.try` to be slow as it takes the default config timeout which is of 30 sec making the test fails very late.
There is an alternate called 

```
   await retry.tryForTime(TIME_IN_MILLISECONDS, async () => {});
```
With this, i limit my test to 5*1000ms which makes failures faster. Its not a hard rule, sometimes tests would need more time, but the default time of 30 seconds definitely makes me feel that something is wrong with the test.